caravel/assets/javascripts/explorev2/components/charts/TimeSeriesLineChart.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { PropTypes } from 'react';
+import * as V from 'victory';
+import theme from '../../../components/VictoryTheme';
+import moment from 'moment';
+import { schemeCategory20c } from 'd3-scale';
+import Legend from './Legend';
+
+const propTypes = {
+  data: PropTypes.array.isRequired,
+  label1: PropTypes.string.isRequired,
+  height: PropTypes.number.isRequired,
+};
+
+export default class TimeSeriesLineChart extends React.Component {
+  constructor(props) {
+    super(props);
+    this.keysToColorsMap = this.mapKeysToColors(props.data);
+  }
+
+  mapKeysToColors(data) {
+    // todo: what if there are more lines than colors in schemeCategory20c?
```

Comment:
also it might be worth checking with @mistercrunch about how he does colors currently. when there are multiple charts on a dashboard and they share some categories, ideally you do not assign/use two different colors for the same category on different charts. at one point he was hashing keys to always get the same color. not that trivial...